### PR TITLE
feat: v2 version of lib

### PR DESCRIPTION
**Pull request by radovan-jorgic**
- Support for the State API
- HTTP client for API requests
- Local development environment creates local artifact files
- Improvements in logging

### CODE REVIEW

File: src/uploader/index.ts [hunk 1]
```diff
@@ -16,17 +16,17 @@ import { Artifact, UploadResponse } from '../types/common';
  * @constructor
  * @param {string} endpoint - The endpoint of the DevRev platform
  * @param {string} token - The token to authenticate with the DevRev platform
+ * @param {boolean} local - A flag to indicate if the uploader should upload to the file-system.
```
Comment (navneel99) [inline src/uploader/index.ts:19]:
```suggestion
 * @param {boolean} local - Flag to indicate the uploader to also upload to the local file system.
```

File: src/index.ts [hunk 2]
```diff
@@ -2,3 +2,6 @@ export * from './adapter';
 export * from './demo-extractor';
 export * from './uploader';
 export * from './types';
+
+export * from './http/client';
+export * from './http/types';
```
Comment (navneel99) [inline src/index.ts:7]: Maybe export just `./http` and export the required in `http/index.ts`

**Review by navneel99 — APPROVED** [review]
Minor comments.

File: src/state/index.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+  if (!STATELESS_EVENT_TYPES.includes(event.payload.event_type)) {
+    console.log('Fetching state');
+    await as.fetchState(newInitialState);
+  }
+
+  return as;
+}
+
+export class AdapterState<ExtractorState> {
```
Comment (samod) [inline src/state/index.ts:21]: Please consider renaming this AdapterState class - it has the same naming as the exported AdapterState type.

**Review by samod — APPROVED** [review]

**Pull request merged**